src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -112,6 +127,57 @@ class RegularSync(
       }
   }
 
+  /**
+    * Handles NewHashesMessege, should only cover this message when we are top of the chain
+    */
+  def handleNewBlockHashesMessages: Receive = {
+    case MessageFromPeer(NewBlockHashes(hashes), peerId) =>
+      val maybePeer = peersToDownloadFrom.find(peer => peer._1.id == peerId)
+      //we allow asking for new hashes when we are not syncing and we can download from specified peer and we are
+      //top of the chain
+      if (notDownloading() && topOfTheChain && maybePeer.isDefined) {
+        val (peer, _) = maybePeer.get
+        val hashesToCheck = hashes.take(syncConfig.maxNewHashes)
+
+        if (!containsAncientBlockHash(hashesToCheck)) {
```

Comment:
If I understand correctly, we're only blacklisting the sender of `NewBlockHashes` if sends us an `ancient` peer. What if it sends us some nonsense hashes (with sensible numbers)? How would it affect our syncing if malicious peers kept sending us such messages causing us to query other peers for blocks they do not know?

Reply:
As I see it if some malicious peer would keep sending us some procured `newBlockHashMessage` (with some valid number )we would ask only this peer for this headers via `GetBlockHeaders` message. And it depends on his answer to this: 

If he would send us some fake headers we would go with our usual strategy ( handle headers and then bodies). It would probably finish with some validation fail during import block and we would blacklist peer. 

If he would respond to us with empty `BlockHeaders` message, we would simply reschedule our normal ask for headers cycle.

So we would never query other peers for some hashes that one malicious peer is sending to us.